api/v1alpha1/mariadb_webhook.go

```diff
@@ -201,3 +203,24 @@ func (r *MariaDB) validatePodDisruptionBudget() error {
 	}
 	return nil
 }
+
+func (r *MariaDB) validateStorage() error {
+
+	// spec.ephemeralStorage and spec.volumeClaimTemplate shall be mutually exclusive
+	if !r.IsEphemeralStorageEnabled() && reflect.ValueOf(r.Spec.VolumeClaimTemplate).IsZero() {
```

Comment:
I think it is better to enforce that both ephemeralStorage and VolumeClaimTemplate are mutually exclusive to avoid confusion.

If neither is configured the pod will never be created with this failure but It could be better to add an explicit error in the webhook.
`  Warning  FailedCreate  7s (x12 over 17s)  statefulset-controller  create Pod mariadb-min-0 in StatefulSet mariadb-min failed error: failed to create PVC storage-mariadb-min-0: PersistentVolumeClaim "storage-mariadb-min-0" is invalid: [spec.accessModes: Required value: at least 1 access mode is required, spec.resources[storage]: Required value]`

 How do you test the webhook + controller locally? I have a port conflict when I launch both. @mmontes11

Reply:
> I think it is better to enforce that both ephemeralStorage and VolumeClaimTemplate are mutually exclusive to avoid confusion.

Makes sense to me as well. When `ephemeralStorage` is `false`, the `volumeClaimTemplate` must be set as well.

> If neither is configured the pod will never be created with this failure but It could be better to add an explicit error in the webhook.

Right, we should explicitly cover this case in the webhook to avoid non meaningful errors for the user like this one:
 `Required value: at least 1 access mode is required`. This would be very hard to debug for the end user.
 
> How do you test the webhook + controller locally? I have a port conflict when I launch both

Probably you have been using the `make webhoo` I am guessing? This target is only intented to spin up a webhook locally to ensure it runs correctly and tthe TLS configuration is fine. The Kubernetes API server is not configured to route request to the local webhook i.e. no `ValidatingWebhookConfiguration` nor `MutatingWebhookConfiguration` are creating pointing to the local webhook, connectivity-wise it would be hard it would be tricky,

Here it is what you can do:
- Build the `mariadb-operator` image and load it in kind. The tag to be built is specified in the Makefile [here](https://github.com/mariadb-operator/mariadb-operator/blob/e654465e675963603159980844fd9f1853cd1c0a/Makefile#L16):
```bash
make docker-dev
```
- Install the operator locally using the helm chart pointing to the previous tag:
```
helm upgrade --install --set image.tag=v0.0.24 mariadb-operator deploy/charts/mariadb-operator
```

This will run and operator and a webhook with your local changes.

Reply:
In any case, please add test cases for the webhook suite covering the ephemeral storage functionality

Reply:
Thanks I was able to valide the webhooks, I have updated the PR description with the errors returned.